kahuna/public/js/search/structured-query/structured-query.js

```diff
@@ -67,6 +68,22 @@ grStructuredQuery.directive('grStructuredQuery', ['subscribe$', function(subscri
 
             subscribe$(scope, ctrl.newQuery$, query => {
                 ngModelCtrl.$setViewValue(query);
+                const structuredQuery = structureQuery(query);
+                structuredQuery.forEach(queryComponent => {
```

Comment:
it looks like this sends an event per chip, but only one for the entire text (and I think none for empty searches), which will overemphasise chip usage. I took a look but also couldn't see a handy way of sending multiple bits of data in a single event. Maybe the best is to create a random uuid for each search and add to each event, which I think might let us correlate all of the events when we query? cc @jonathonherbert for any ideas

Reply:
Interesting idea. Makes me think - if we had session ID and user ID as well we'd have the potential to do some decent analysis down the line.

Reply:
Sending chips individually was the intentional behaviour but I think your suggestion makes more sense.

Reply:
We now should have all three. Will do some testing to check it's working in the morning